nomad/node_endpoint.go

```diff
@@ -1136,6 +1151,32 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 	return nil
 }
 
+// unclaimVolumesForTerminalAllocs unpublishes and unclaims CSI volumes
+// that belong to the alloc if it is terminal.
+func (n *Node) unclaimVolumesForTerminalAllocs(args *structs.AllocUpdateRequest, alloc *structs.Allocation, taskGroup *structs.TaskGroup) error {
+	for _, volume := range taskGroup.Volumes {
+
+		// TODO(tgross): we also need to call ControllerUnpublishVolume CSI RPC here
+		// but the server-side CSI client + routing hasn't been implemented yet
+
+		req := &structs.CSIVolumeClaimRequest{
+			VolumeID:     volume.Name,
```

Comment:
I'm not totally sure if I should be checking `volume.Name` or `volume.Source` here?

Reply:
Good catch, it should be `volume.Source`

Reply:
Fixed in 09a7b6a